apps/dg/utilities/data_utilities.js

```diff
@@ -173,4 +173,96 @@ DG.DataUtilities.toString = function (iValue) {
         return String(iValue);
     }
     return "";
-};
\ No newline at end of file
+};
+
+var kTypeNumber = 1,
+    kTypeBoolean = 2,
+    kTypeString = 3,
+    kTypeNaN = 4,
+    kTypeError = 5,
+    kTypeSimpleMap = 6, // e.g. boundaries
+    kTypeUnknown = 7,
+    kTypeEmpty = 8;
+
+/**
+  Returns the type code corresponding to the specified value.
+ */
+DG.DataUtilities.typeCode = function(value) {
```

Comment:
Should DG.DataUtilitities.typeCode have specific handling for Date objects? In the "Four Seals" std document, the "date" field does not sort.